contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -1641,3 +1644,55 @@ truncate_tsql_identifier(char *ident)
 					  PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
 
 }
+
+void
+rewrite_call_stmt(CallStmt *callstmt)
+{
+	PLExecStateCallStack *top_es_entry = NULL;
+	List*                name;
+	char                 *dbname = NULL;
+
+	if (exec_state_call_stack)
+		top_es_entry = exec_state_call_stack->next;
+
+	while (top_es_entry != NULL)
+	{
+		if (top_es_entry->estate && top_es_entry->estate->err_stmt &&
+			top_es_entry->estate->err_stmt->cmd_type == PLTSQL_STMT_EXEC &&
+			top_es_entry->estate->db_name != NULL)
+		{
+			dbname = pstrdup(top_es_entry->estate->db_name);
+			break;
+		}
+		top_es_entry = top_es_entry->next;
+	}
```

Comment:
Does it make sense to update current es_entry->estate->db_name with dbname that we fetched by looping over all parent entries?
That will save us from iterating over whole es stack for every CALL stmt.